build/tools/Automation/DotNet/DotNet.cs

```diff
@@ -77,13 +81,13 @@ public override async Task<List<PlatformConstant>> GetPlatformConstantsAsync(str
                 {
                     // check releasedToday again since there
                     // are still releases from other dates.
-                    if (!DatesMatch(dateTarget, release.ReleaseDate))
+                    string sdkVersion = release.Sdk.Version;
+                    if (!DatesMatch(dateTarget, release.ReleaseDate) || this.prodSdkVersions.Contains(sdkVersion))
```

Comment:
We still need to check if the `dateTarget` and `ReleaseDate` match otherwise really old SDKs that are not in our storage account (like .NET2.1) get added.